bigframes/core/blocks.py

```diff
@@ -899,7 +901,22 @@ def aggregate_all_and_stack(
             index_col_ids = [
                 guid.generate_guid() for i in range(self.column_labels.nlevels)
             ]
-            result_expr = self.expr.aggregate(aggregations, dropna=dropna).unpivot(
+            aggregate_expr = self.expr.aggregate(aggregations, dropna=dropna)
+
+            # if all aggregates are of the same dtype then we should set that
+            # dtype in the result
+            if dtype is None:
+                aggregate_dtypes = set(
+                    [
+                        aggregate_expr.get_column_type(c)
+                        for c in aggregate_expr.column_ids
+                    ]
+                )
+                if len(aggregate_dtypes) == 1:
+                    dtype = aggregate_dtypes.pop()
+            if dtype is None:
+                dtype = pd.Float64Dtype()
+            result_expr = aggregate_expr.unpivot(
```

Comment:
Can we use some variant of this existing logic: https://github.com/googleapis/python-bigquery-dataframes/blob/main/bigframes/dtypes.py#L657 to find the common supertype?

Reply:
Done, PTAL, thanks!